### ISSUE TITLE
Fullscreen mode only works for first video

### ISSUE THREAD

**Issue by sebk-666** [iina/iina#4202]
**System and IINA version:**
- macOS 13.2
- IINA 1.3.1 Build 133, mpv 0.34.1

**Expected behavior:**
I have set "General / When Media is Opened / Enter Fullscreen", "Always open media in new window".
When clicking on a video file, I want IINA to play it back in fullscreen mode. Afterwards IINA should close the window, but the application should not quit.
When clicking on another video (or the same one again), IINA should play it back in fullscreen mode, close the window at the end and stay open.
Repeat.

**Actual behavior:**
Clicking on the initial video IINA opens and plays the video in fullscreen mode. The window closes at the end. IINA stays open.
Clicking on another/the same video again, IINA starts playing back the video in windowed mode.

I was not able to find a way to playback each video in fullscreen mode (unless "quit after all windows are closed" is active).
Is this the intended behavior?

- [x] MPV does not have this problem.

Consecutive executions of "`mpv /path/to/video.mp4`" always play back in fullscreen mode.

**How often does this happen?**
Every time.
[iina.log](https://github.com/iina/iina/files/10536027/iina.log)
[mpv.log](https://github.com/iina/iina/files/10536028/mpv.log)

**Comment by low-batt** [comment]
Reproduced for me.

The  code supporting this setting is found in `PlayerCore.fileLoaded`:
```swift
// set initial properties for the first file
if info.justLaunched {
  if Preference.bool(for: .fullScreenWhenOpen) && !mainWindow.fsState.isFullscreen && !isInMiniPlayer {
    DispatchQueue.main.async(execute: self.mainWindow.toggleWindowFullScreen)
  }
  info.justLaunched = false
}
```

As the `Enter fullscreen` setting is listed under `When media is opened` I would expect it to take effect every time a file is opened. Yet as you can see above it is specifically coded to only do this "for the first file".

I will need to check with others as to why that is.

**Comment by low-batt** [comment]
The feedback I received was that this setting should put the player into full screen mode for each video played. That means the reported behavior is incorrect.

The proposed fix merely removes the code that limits this setting to the first video. That code was added in commit ad3d6b68d4aee65d712c70f385edde2b6cf92c90 back in 2017.

I'm a bit worried there is some reason for that code that I've missed, but the setting worked as expected when I tested the above change.

**Comment by sebk-666** [comment]
I'm just guessing here, but it's possible that in case multiple files were opened to be played in a row and the user changing the mode from fullscreen to window during playback of one of the files, it would change back to fullscreen when the next video opened and this behavior might have confused some users.

I only had a quick glimpse at the code (knowing nothing about Swift), but the 2017 commit to me only looks like a code cleanup, ditching the use of `needEnterFullScreenForNextMedia` (defined in the same file) for `info.justLaunched` (defined in PlaybackInfo.swift).
Note that there also would have been a `info.justOpenedFile` which to me looks more suitable in light of this issue and it appears that the choice to change to fullscreen only on the first file was indeed made intentionally.

It's also possible that the state of the `needEnterFullScreenForNextMedia` and `info.justLaunched` variables is not the same with all the use cases.

Maybe @lhc70000 could shed some light on this.

**Comment by low-batt** [comment]
Yes, it could be that users were confused as to why IINA unexpectedly went into full screen mode on them, forgetting that they enabled this setting. If that was the issue I'm not sure how I would have addressed it. But based on the wording in the settings the current behavior seems wrong to me.

On the referenced commit, agreed, that may just be a code cleanup. As that was back in 2017 I did not trace the history any further.

One of the senior developers will review the changes before they are merged. I did check in with one developer before proposing this change. Of course the pull request still might be rejected during review.
